src/vellum/workflows/integrations/tests/test_vellum_integration_service.py

```diff
@@ -1,225 +1,206 @@
 import pytest
 from unittest import mock
 
+from vellum.workflows.constants import VellumIntegrationProviderType
 from vellum.workflows.exceptions import NodeException
 from vellum.workflows.integrations.vellum_integration_service import VellumIntegrationService
+from vellum.workflows.types.definition import VellumIntegrationToolDetails
 
 
-def test_vellum_integration_service_get_tool_definition_success():
+def test_vellum_integration_service_get_tool_definition_success(vellum_client):
     """Test that tool definitions are successfully retrieved from Vellum API"""
-    with mock.patch(
-        "vellum.workflows.integrations.vellum_integration_service.create_vellum_client"
-    ) as mock_create_client:
-        # GIVEN a mock Vellum client configured to return a tool definition
-        mock_client = mock.MagicMock()
-        mock_create_client.return_value = mock_client
-
-        mock_response = mock.MagicMock()
-        mock_response.name = "GITHUB_CREATE_AN_ISSUE"
-        mock_response.description = "Create a new issue in a GitHub repository"
-        mock_response.parameters = {
-            "type": "object",
-            "properties": {
-                "repo": {"type": "string", "description": "Repository name"},
-                "title": {"type": "string", "description": "Issue title"},
-                "body": {"type": "string", "description": "Issue body"},
-            },
-            "required": ["repo", "title"],
-        }
-        mock_response.provider = "COMPOSIO"
-
-        mock_client.integrations.retrieve_integration_tool_definition.return_value = mock_response
-
-        # WHEN we request a tool definition
-        service = VellumIntegrationService()
-        result = service.get_tool_definition(
-            integration="GITHUB",
-            provider="COMPOSIO",
-            tool_name="GITHUB_CREATE_AN_ISSUE",
-        )
+    mock_client = vellum_client
+    mock_client.integrations = mock.MagicMock()
+
+    mock_response = mock.MagicMock()
+    mock_response.name = "GITHUB_CREATE_AN_ISSUE"
+    mock_response.description = "Create a new issue in a GitHub repository"
+    mock_response.parameters = {
+        "type": "object",
+        "properties": {
+            "repo": {"type": "string", "description": "Repository name"},
+            "title": {"type": "string", "description": "Issue title"},
+            "body": {"type": "string", "description": "Issue body"},
+        },
+        "required": ["repo", "title"],
+    }
+    mock_response.provider = "COMPOSIO"
+
+    mock_client.integrations.retrieve_integration_tool_definition.return_value = mock_response
```

Comment:
[nit] preference for setting the right hand side to the actual pydantic model that this response returns rather than a magic mock